Configure rehypeAutoLinkHeadings + add className to a tag for headings

glitch-txs opened the pull request:
Seems like rehypeAutoLinkHeadings needed the `behavior: 'wrap'` option in order to work.

I additionally added a `className` called `heading` (which is optional) to try to overide the default anchor style values that seem to be set globally in [this part](https://github.com/solidjs/solid-docs-next/blob/81ed9b969667577c8b903cb8cdef3fdc95bc1798/src/ui/markdown-components.tsx#L32), my intention was to keep the previous styling for the titles but feel free to remove or change that, it still has the underline and hover effect.

![image](https://github.com/solidjs/solid-docs-next/assets/66949816/93ca87e4-f701-4e06-9a1e-cadd20ea17aa)

closes https://github.com/solidjs/solid-docs-next/issues/427

atilafassina commented on src/styles.css:
```diff
@@ -15,6 +15,10 @@ html {
 	.prose h2 {
 		@apply mt-0;
 	}
+  
+  .heading{
+    color: #F1F5F9 !important;
+  }
```
because of the `!important` , I believe this belongs in the `utilities` layer, not `base`.

Would you mind moving it there?

[tailwind layers](https://tailwindcss.com/docs/adding-custom-styles#adding-custom-utilities)

glitch-txs commented:
Thanks @atilafassina for the review and feedback! I put a little bit more of effort to improve the styling.

- Added the # symbol on hover (most docs do this)
- removed unnecessary style

![image](https://github.com/solidjs/solid-docs-next/assets/66949816/49cd500c-d441-4596-8b86-e388c62db6b8)
